### ISSUE TITLE
hostname ignored in Net::Whois::Generic

### ISSUE THREAD

**Issue by ssinyagin** [arhuman/Net-Whois-RIPE#20]
hi,
Net::Whois::Generic->query() always calls adapt_query(), which always calls _find_rir. As a result, the "hostname" attribute in the Net::Whois::Generic object is completely ignored.

So, there should be a way to stop using _find_rir if we want to use only a specific WHOIS server.

The issue popped up because the local DNS server limits the rate of requests, and my script fails even if I specify the IP address of whois.ripe.net.

thanks

**Comment by arhuman** [comment]
May I consider this issue as fixed ?

**Comment by ssinyagin** [comment]
I guess so, thanks

**Comment by ssinyagin** [comment]
hi, there's again trouble with using "hostname" attribute. For example, AS11344 is not present in RIPE, but is present in RADB. The following script produces an empty list of results, because your module still tries to access RIPE:

```
use strict;
use warnings;
use Net::Whois::Generic;

my $w = Net::Whois::Generic->new('hostname' => 'whois.radb.net');

my ($as_found) = $w->query('-T aut-num AS11344',
                           { type => 'AutNum' });

print $as_found->as_name();
```

It looks like the constructor ignored "hostname" completely because it's not in %default_options hash.

**Comment by arhuman** [comment]
Fixed in 2.006001
